Content.Server/Tools/ToolSystem.Tile.cs

```diff
@@ -0,0 +1,116 @@
+using System.Threading;
+using Content.Server.Fluids.Components;
+using Content.Server.Tools.Components;
+using Content.Shared.Interaction;
+using Content.Shared.Maps;
+using Content.Shared.Tools.Components;
+using Content.Shared.Database;
+using Robust.Shared.Map;
+
+namespace Content.Server.Tools;
+
+public sealed partial class ToolSystem
+{
+    private void InitializeToolWithTile()
+    {
+        SubscribeLocalEvent<ToolWorksWithTilesComponent, AfterInteractEvent>(OnToolWithTileAfterInteract);
+        SubscribeLocalEvent<ToolWorksWithTilesComponent, TileToolCompleteEvent>(OnToolWithTileComplete);
+        SubscribeLocalEvent<ToolWorksWithTilesComponent, TileToolCancelledEvent>(OnToolWithTileCancelled);
+    }
+
+    private void OnToolWithTileCancelled(EntityUid uid, ToolWorksWithTilesComponent component, TileToolCancelledEvent args)
+    {
+        component.CancelTokenSource = null;
+    }
+
+    private void OnToolWithTileComplete(EntityUid uid, ToolWorksWithTilesComponent component, TileToolCompleteEvent args)
+    {
+        component.CancelTokenSource = null;
+        var gridUid = args.Coordinates.GetGridUid(EntityManager);
+        if (!_mapManager.TryGetGrid(gridUid, out var grid))
+        {
+            Logger.Error("Attempted use tool on a non-existent grid?");
+            return;
+        }
+
+        if (!TryComp(component.Owner, out ToolComponent? tool))
+            return;
+
+        var tileRef = grid.GetTileRef(args.Coordinates);
+
+        if (component.RequiresUnobstructed && tileRef.IsBlockedTurf(true))
+            return;
+        
+        if (tileRef.TryDeconstructWithToolQualities(tool.Qualities, _mapManager, _tileDefinitionManager, EntityManager))
```

Comment:
this function should just get moved to being in the same system, i doubt it will be called elsehwere? (also the other tileref methods in this same vein were removed in a diff PR)